python/ray/serve/controller.py

```diff
@@ -519,6 +534,27 @@ async def __init__(self,
 
         asyncio.get_event_loop().create_task(self.run_control_loop())
 
+    async def wait_for_event(self, uuid: ResultId) -> FutureResult:
+        future = self.inflight_results[uuid]
+        result = await future
+        # NOTE(ilr) We have to delete this *after* we return to avoid a
+        # situation where the controller crashes *after* deleting and before
+        # returning to the API (and user).
+        asyncio.get_event_loop().call_soon(
+            lambda: self.inflight_results.pop(uuid))
+        return result
+
+    def _create_and_set_result(self, result: bool,
```

Comment:
I didn't get what does this function do first, how about:
- `_create_event_with_result(self, goal_state: Dict[str, Any],  is_success: bool)`

Reply:
This is just a temporary helper that'll be removed in a subsequent PR, right? In that case not too concerned about naming. That said, I also found it somewhat confusing. Maybe `create_event_and_set_result`.

Reply:
I can switch to that! It will be removed after this is properly set by the actual result.